src/lib/wicked/files.js

```diff
@@ -26,20 +26,41 @@
  */
 
 import cockpit from 'cockpit';
+import bootProtocol from '../model/bootProtocol';
 
 /**
  * @ignore
  */
 const connectionToSysconfig = (connection) => {
     return {
         NAME: connection.name,
-        BOOTPROTO: connection.bootProto,
+        BOOTPROTO: bootProtoFor(connection),
         STARTMODE: connection.startMode,
         ...bridgeToSysconfig(connection.bridge),
-        ...bondToSysconfig(connection.bond)
+        ...bondToSysconfig(connection.bond),
+        ...addressesToSysconfig(connection)
     };
 };
 
+const bootProtoFor = (connection) => {
+    const {
+        ipv4: { bootProto: ipv4Proto, addresses: ipv4Addresses },
+        ipv6: { bootProto: ipv6Proto, addresses: ipv6Addresses }
+    } = connection;
```

Comment:
Nice _nested_ destructuring. Although it is a bit complex to catch if you are not familiar with it.